### PR TITLE
formula_installer: improve no-bottle error message

### LINKED ISSUE #10180: No clear message given when CLT is needed for a bottle
# Bug report

**Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.**

- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [ ] ~~ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?~~
```% brew gist-logs python
Error: No logs.
```
- [x] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

```
% brew doctor
Your system is ready to brew.
```

```
% brew config
HOMEBREW_VERSION: 2.7.1-54-g8945803
ORIGIN: https://github.com/Homebrew/brew
HEAD: 8945803432764e6941ebbb55485fdf346d640ad0
Last commit: 4 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: f2cc04c45e0fde4973784f5376e061aa0ccff24a
Core tap last commit: 47 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit haswell
Clang: 12.0 build 1200
Git: 2.24.3 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.1-x86_64
CLT: N/A
Xcode: 12.3
```

## What you were trying to do (and why)

Install python. The existing python install is the mac built-in one, I think.

## What happened (include command output)

<details>
  <summary>brew install python</summary>
  <pre>
Error: python@3.9: no bottle available!
You can try to install from source with e.g.
  brew install --build-from-source python@3.9
Please note building from source is unsupported. You will encounter build
failures with some formulae. If you experience any issues please create pull
requests instead of asking for help on Homebrew's GitHub, Twitter or any other
official channels.
  </pre>
</details>

## What you expected to happen

I expect there to be a bottle available to use - at least https://formulae.brew.sh/formula/python@3.9 suggests there's a bottle available for big sur.

## Step-by-step reproduction instructions (by running `brew install` commands)

See above.

### PR DESCRIPTION
**Pull request by carlocab**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?
    - This returned a `java_spec` error and a `tap-new_spec` error. `brew tests` produces these errors even when run on the master branch.
- [x] Have you successfully run `brew man` locally and committed any changes?

-----

Closes #10180.

This PR changes the error message when you try to install `python@3.9` and you have no CLT installed to:
```
❯ HOMEBREW_NO_AUTO_UPDATE=1 brew install python@3.9
Error: python@3.9: unable to pour bottle!
The bottle needs the Apple Command Line Tools to be installed.
  You can install them, if desired, with:
    xcode-select --install
You can try to install from source with e.g.
  brew install --build-from-source python@3.9
Please note building from source is unsupported. You will encounter build
failures with some formulae. If you experience any issues please create pull
requests instead of asking for help on Homebrew's GitHub, Twitter or any other
official channels.
```
If the formula does not specify a `pour_bottle?` method but also has no bottles:
```
❯ HOMEBREW_NO_AUTO_UPDATE=1 brew install a2ps
Error: a2ps: unable to pour bottle!
You can try to install from source with e.g.
  brew install --build-from-source a2ps
Please note building from source is unsupported. You will encounter build
failures with some formulae. If you experience any issues please create pull
requests instead of asking for help on Homebrew's GitHub, Twitter or any other
official channels.
```
I suspect there may be more refactoring in order for the `pour_bottle?` method in `formula_installer.rb`, but I'd prefer to address those later.

### CODE REVIEW

**Review by Rylan12 — COMMENTED** [review]
I think the `no bottle available!` message is a tad misleading in this case. Maybe instead: `bottle conditions not met!`. No bottle available feels more "final" as if it's saying "sorry there's nothing you can do except build from source" while this is more like "before you can use a bottle you must do this".

Maybe that's not the case for all formulae that use `pour_bottle?`, though.

**Comment by carlocab** [comment]
How about `no suitable bottle available`?

**Review by Rylan12 — COMMENTED** [review]
Sorry, wasn't clear with what I was thinking.

I was thinking this "unable to pour bottle" message would be _only_ for instances where `formula.pour_bottle?` is false.

When this message is showing in situations like no ARM bottles are available, I think it makes sense to still have the "no bottle available" message. That's a little more accurate to what the exact issue is for the user, I think.

**Comment by carlocab** [comment]
I agree, but I'm still mulling over the best way to implement it.

For instance, maybe it's enough to just add `&& formula.pour_bottle?` to the conditions required by the `if` statement and drop any of the changes I've made here so far.

**Comment by Rylan12** [comment]
Ah, you know what? I think this is what you want to do instead:

Change the `!pour_bottle?` condition on line 244 (just above you changes) to `!pour_bottle(output_warning: true)`

You may also need to adjust the messaging. See if that gives a better result. `pour_bottle?` is already calling `formula.pour_bottle?` and it displays the message there.

**Comment by carlocab** [comment]
I've considered that too, but decided against it since adjusting the messaging already in the `pour_bottle?` method seemed overly complicated... but I'll think about it again.

**Comment by carlocab** [comment]
Thinking about it some more, I don't think I should change the messaging in the `pour_bottle?` method, since that has implications on Linux as well. Whatever changes need to be made will have to be where I've already proposed them in this PR. Still thinking about the right way to do it though.

File: Library/Homebrew/formula_installer.rb [hunk 1]
```diff
@@ -242,15 +242,21 @@ def check_install_sanity
        # Integration tests override homebrew-core locations
        ENV["HOMEBREW_TEST_TMPDIR"].nil? &&
        !pour_bottle?
-      raise CannotInstallFormulaError, <<~EOS
-        #{formula}: no bottle available!
+      message = <<~EOS
+        #{formula}: unable to pour bottle!
```
Comment (MikeMcQuaid) [inline Library/Homebrew/formula_installer.rb:246]:
```suggestion
        #{formula}: no bottle available!
```

Let's leave this as-is for now and merge as-is and then can do more in a follow-up PR.

**Review by MikeMcQuaid — APPROVED** [review]

**Review by MikeMcQuaid — APPROVED** [review]

**Pull request merged**